ext/bcmath/tests/bcround.phpt

```diff
@@ -0,0 +1,851 @@
+}
+echo "\n";
+
+$not_boundary_value_cases = [
+    ['1.1', 0],
+    ['1.2', 0],
+    ['1.3', 0],
+    ['1.4', 0],
+    ['1.6', 0],
+    ['1.7', 0],
+    ['1.8', 0],
+    ['1.9', 0],
+    ['-1.1', 0],
+    ['-1.2', 0],
+    ['-1.3', 0],
+    ['-1.4', 0],
+    ['-1.6', 0],
+    ['-1.7', 0],
+    ['-1.8', 0],
+    ['-1.9', 0],
+];
+
+echo "========== not boundary values ==========\n";
+foreach ($modes as $mode) {
+    echo "------ {$mode} ------\n";
+    foreach ($not_boundary_value_cases as [$num, $precision]) {
+        $printResult($num, $precision, 20, $mode);
+    }
+    echo "\n";
+}
+
+$minus_precision_cases = [
+    ['1230', -1],
+    ['1235', -1],
+    ['-1230', -1],
+    ['-1235', -1],
+    ['3400.0000', -2],
+    ['3400.0001', -2],
+    ['3450.0000', -2],
+    ['3450.0001', -2],
+    ['-3400.0000', -2],
+    ['-3400.0001', -2],
+    ['-3450.0000', -2],
+    ['-3450.0001', -2],
+];
+
+echo "========== minus precision ==========\n";
+foreach ($modes as $mode) {
+    echo "------ {$mode} ------\n";
+    foreach ($minus_precision_cases as [$num, $precision]) {
+        $printResult($num, $precision, 20, $mode);
+    }
+    echo "\n";
+}
+
+$zero_precision_cases = [
+    ['1235', 0],
+    ['1235.0', 0],
+    ['1235.000001', 0],
+    ['1235.5', 0],
+    ['1235.500001', 0],
+    ['-1235', 0],
+    ['-1235.0', 0],
+    ['-1235.000001', 0],
+    ['-1235.5', 0],
+    ['-1235.500001', 0],
+    ['0.0001', 0],
+    ['0.5', 0],
+    ['0.5000', 0],
+    ['0.5001', 0],
+    ['-0.0001', 0],
+    ['-0.5', 0],
+    ['-0.5000', 0],
+    ['-0.5001', 0],
+];
+
+echo "========== zero precision ==========\n";
+foreach ($modes as $mode) {
+    echo "------ {$mode} ------\n";
+    foreach ($zero_precision_cases as [$num, $precision]) {
+        $printResult($num, $precision, 20, $mode);
+    }
+    echo "\n";
+}
+
+$plus_precision_cases = [
+    ['28.40', 1],
+    ['28.4000001', 1],
+    ['28.45', 1],
+    ['28.4500001', 1],
+    ['-28.40', 1],
+    ['-28.4000001', 1],
+    ['-28.45', 1],
+    ['-28.4500001', 1],
+    ['153.90', 1],
+    ['153.9000001', 1],
+    ['153.95', 1],
+    ['153.9500001', 1],
+    ['-153.90', 1],
+    ['-153.9000001', 1],
+    ['-153.95', 1],
+    ['-153.9500001', 1],
+    ['0.000001', 3],
+    ['0.0005', 3],
+    ['0.000500', 3],
+    ['0.000501', 3],
+    ['-0.000001', 3],
+    ['-0.0005', 3],
+    ['-0.000500', 3],
+    ['-0.000501', 3],
+];
+
+echo "========== plus precision ==========\n";
+foreach ($modes as $mode) {
+    echo "------ {$mode} ------\n";
+    foreach ($plus_precision_cases as [$num, $precision]) {
+        $printResult($num, $precision, 20, $mode);
+    }
+    echo "\n";
+}
+
+echo "========== value error ==========\n";
+try {
+    bcround('hoge');
+} catch (Throwable $e) {
+    echo $e->getMessage()."\n";
+}
+
+try {
+    bcround('0.00.1');
+} catch (Throwable $e) {
+    echo $e->getMessage()."\n";
+}
+
+try {
+    bcround('0.001', 0, 1000);
+} catch (Throwable $e) {
+    echo $e->getMessage()."\n";
+}
```

Comment:
Please split all the different sections into different test files. This makes it easier to review and figure out what each test is for.